microsoft/validation/k_inference_js.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+import pandas as pd
+import json
+from sklearn.metrics import confusion_matrix, precision_score, recall_score, f1_score, accuracy_score
+import librosa
+from Config import Config
+from keras import losses, models, optimizers
+from keras.activations import relu, softmax
+from keras.callbacks import (EarlyStopping, LearningRateScheduler,
+                             ModelCheckpoint, TensorBoard, ReduceLROnPlateau)
+from keras.layers import (Convolution2D, GlobalAveragePooling2D, BatchNormalization, Flatten,
+                          GlobalMaxPool2D, MaxPool2D, concatenate, Activation, Input, Dense)
+from keras.utils import Sequence, to_categorical
+from keras import backend as K
+import argparse
+
+parser = argparse.ArgumentParser(
+    'Infer. ex: python inference_2dcnn --model_folder="2dcnn/" --wav_file="input/freesound-audio-tagging/shuffled/test/deadpool1_00-01-30.000.wav" --json_filepath="infer.txt"')
+parser.add_argument(
+    "--model_folder", help="directory to store models", type=str, default=R'2dcnn/')
+parser.add_argument("--wav_file", help="path to a wav file", type=str,
+                    default=R'input/freesound-audio-tagging/shuffled/test/deadpool1_00-01-30.000.wav')
+parser.add_argument(
+    "--json_filepath", help="directory to store predictions", type=str, default=R'infer.txt')
+args = parser.parse_args()
+
+
+def get_2d_conv_model(config):
+
+    nclass = config.n_classes
+
+    inp = Input(shape=(config.dim[0], config.dim[1], 1))
+    x = Convolution2D(32, (4, 10), padding="same")(inp)
+    x = BatchNormalization()(x)
+    x = Activation("relu")(x)
+    x = MaxPool2D()(x)
+
+    x = Convolution2D(32, (4, 10), padding="same")(x)
+    x = BatchNormalization()(x)
+    x = Activation("relu")(x)
+    x = MaxPool2D()(x)
+
+    x = Convolution2D(32, (4, 10), padding="same")(x)
+    x = BatchNormalization()(x)
+    x = Activation("relu")(x)
+    x = MaxPool2D()(x)
+
+    x = Convolution2D(32, (4, 10), padding="same")(x)
+    x = BatchNormalization()(x)
+    x = Activation("relu")(x)
+    x = MaxPool2D()(x)
+
+    x = Flatten()(x)
+    x = Dense(64)(x)
+    x = BatchNormalization()(x)
+    x = Activation("relu")(x)
+    out = Dense(nclass, activation=softmax)(x)
+
+    model = models.Model(inputs=inp, outputs=out)
+    opt = optimizers.Adam(config.learning_rate)
+
+    model.compile(
+        optimizer=opt, loss=losses.categorical_crossentropy, metrics=['acc'])
+    return model
+
+
+config = Config(sampling_rate=44100, audio_duration=2, n_folds=10,

```

Comment:
All of this should be in the `if __name__ == "__main__":` as well. Pull down the `argparser` stuff down to here and in put it all in one big `if __name__ == "__main__":`

Reply:
Working!